modules/phase_field/src/kernels/ACInterfaceKobayashi1.C

```diff
@@ -0,0 +1,120 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ACInterfaceKobayashi1.h"
+
+template<>
+InputParameters validParams<ACInterfaceKobayashi1>()
+{
+  InputParameters params = validParams<KernelGrad>();
+  params.addClassDescription("Anisotropic gradient energy Allen-Cahn Kernel Part 1");
+  params.addParam<MaterialPropertyName>("mob_name", "L", "The mobility used with the kernel");
+  params.addParam<MaterialPropertyName>("eps_name", "eps", "The anisotropic parameter");
+  params.addParam<MaterialPropertyName>("eps1_name", "eps1", "The derivative of anisotropic parameter with respect to angle");
+  params.addCoupledVar("args", "Vector of nonlinear variable arguments this object depends on");
+  return params;
+}
+
+ACInterfaceKobayashi1::ACInterfaceKobayashi1(const InputParameters & parameters) :
+    DerivativeMaterialInterface<JvarMapInterface<KernelGrad> >(parameters),
+    _L(getMaterialProperty<Real>("mob_name")),
+    _dLdop(getMaterialPropertyDerivative<Real>("mob_name", _var.name())),
+    _eps(getMaterialProperty<Real>("eps_name")),
+    _eps1(getMaterialProperty<Real>("eps1_name"))
+
+{
+  // Get number of coupled variables
+  unsigned int nvar = _coupled_moose_vars.size();
+
+  // reserve space for derivatives
+  _dLdarg.resize(nvar);
+  _depsdarg.resize(nvar);
+  _deps1darg.resize(nvar);
+
+  // Iterate over all coupled variables
+  for (unsigned int i = 0; i < nvar; ++i)
+  {
+    _dLdarg[i] = &getMaterialPropertyDerivative<Real>("mob_name", _coupled_moose_vars[i]->name());
+    _depsdarg[i] = &getMaterialPropertyDerivative<Real>("eps_name", _coupled_moose_vars[i]->name());
+    _deps1darg[i] = &getMaterialPropertyDerivative<Real>("eps1_name", _coupled_moose_vars[i]->name());
+  }
+}
+
+RealGradient
+ACInterfaceKobayashi1::precomputeQpResidual()
+{
+  // Set interfacial part of residual
+  RealVectorValue v = _eps[_qp]* _eps1[_qp] * _L[_qp] * _grad_u[_qp];
+  std::swap(v(0), v(1));
+  v(0) *= -1.0;
+  return v;
+}
+
+RealGradient
+ACInterfaceKobayashi1::precomputeQpJacobian()
+{
+  Real cos_AC1;
+  Real depsdop_AC1;
+  Real deps1dop_AC1;
```

Comment:
I agree, I thought I had changed those. There might be an unpushed commit at my machine at home :-/